python/ray/data/dataset.py

```diff
@@ -286,6 +286,7 @@ def map(
         fn: UserDefinedFunction[Dict[str, Any], Dict[str, Any]],
         *,
         compute: Optional[ComputeStrategy] = None,
+        fn_constructor_args: Optional[Iterable[Any]] = None,
```

Comment:
should also add `fn_constructor_kwargs`?

Reply:
I prefer us to do the minimal we need here. We can add later if there's ask.